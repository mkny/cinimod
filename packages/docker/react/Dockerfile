FROM node:14-alpine as builder

WORKDIR /app

COPY . .

ENV CI=true \
    GENERATE_SOURCEMAP=false

RUN yarn --check-files &&\
  yarn build

FROM nginx:stable-alpine

COPY --from=builder /app/build/ /usr/share/nginx/html

COPY --from=builder /app/node_modules/@cinimod/docker/react/nginx.conf /etc/nginx/conf.d/default.conf
