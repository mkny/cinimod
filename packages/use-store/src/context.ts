import { createContext, useContext } from "react";

import { IContext } from "./types";

const storeContext = createContext<IContext>({
	state: {},
	set: () => void 0,
	get: (): any => void 0,
	dispatch: () => {}
});

const { Provider: StoreProvider } = storeContext;

const useStoreContext = () => useContext(storeContext);

export default storeContext;
export { StoreProvider, useStoreContext };
