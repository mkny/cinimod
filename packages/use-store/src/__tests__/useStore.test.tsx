import React from "react";
import { act, renderHook } from "@testing-library/react-hooks";
import { render, screen } from "@testing-library/react";

import { withStore, Provider, useStore, useStoreContext } from "../index";
import { MISSING_CONTEXT_ERROR } from "../context";

const MockedComponent = (props: any) => <div {...props} />;

describe("useStore test suite", () => {
	it("should init with default values", () => {
		const { result } = renderHook(() => useStore(), {
			wrapper: withStore({
				initialState: {
					label: "value"
				}
			})(MockedComponent)
		});

		expect(result.current.state).toStrictEqual({ label: "value" });
	});
	it("should be able to set a value", () => {
		const { result } = renderHook(() => useStore(), {
			wrapper: withStore({})(MockedComponent)
		});

		act(() => {
			result.current.set("xabl", "au");
		});

		expect(result.current.state).toStrictEqual({ xabl: "au" });
	});
	it("should be able to useStore with initialPlacement", () => {
		const { result } = renderHook(() => useStore("xab"), {
			wrapper: withStore({
				initialState: {
					xab: "lau"
				}
			})(MockedComponent)
		});

		expect(result.current.state).toStrictEqual("lau");
	});
	it("should be able to set a value and get rootState", () => {
		const { result } = renderHook(() => useStore("xab"), {
			wrapper: withStore({
				initialState: {
					label: "value"
				}
			})(MockedComponent)
		});

		act(() => {
			result.current.set("lau");
		});

		expect(result.current.get()).toStrictEqual("lau");
		expect(result.current.rootState).toStrictEqual({
			label: "value",
			xab: "lau"
		});
	});
	it("should be able to set a object", () => {
		const { result } = renderHook(() => useStore(), {
			wrapper: withStore({
				initialState: {
					mykeyDefault: "myValueDefault"
				}
			})(MockedComponent)
		});

		act(() => {
			result.current.set({
				mykey1: "myValue1",
				mykey2: "myValue2"
			});
		});

		expect(result.current.get()).toStrictEqual({
			mykeyDefault: "myValueDefault",
			mykey1: "myValue1",
			mykey2: "myValue2"
		});
		expect(result.current.get("mykey1")).toStrictEqual("myValue1");
	});
	it("should be able to set and get with placement", () => {
		const { result } = renderHook(() => useStore("login"), {
			wrapper: withStore({})(MockedComponent)
		});

		act(() => {
			result.current.set("xab", "lau");
		});

		expect(result.current.get(["xab"])).toStrictEqual("lau");
		expect(result.current.get("xab")).toStrictEqual("lau");
		expect(result.current.get()).toStrictEqual({ xab: "lau" });
	});
	it("should be able to get default values (without provider)", () => {
		const { result } = renderHook(() => useStoreContext());

		expect(() => result.current.dispatch()).toThrow(MISSING_CONTEXT_ERROR);
		expect(() => result.current.get()).toThrow(MISSING_CONTEXT_ERROR);
		expect(() => result.current.set({})).toThrow(MISSING_CONTEXT_ERROR);
	});
	it("should be able to snapshot", () => {
		const MyComponent = () => {
			const { state } = useStoreContext();
			return <h1>{state.label}</h1>;
		};
		const { container } = render(
			<Provider
				initialState={{
					label: "value"
				}}
			>
				<MyComponent />
			</Provider>
		);

		expect(screen.getByText(/value/)).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});
});
