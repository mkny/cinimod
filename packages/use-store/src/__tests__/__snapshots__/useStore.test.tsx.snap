// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useStore test suite should be able to snapshot 1`] = `
<h1>
  value
</h1>
`;
